src/main/java/org/gaul/s3proxy/azureblob/AzureBlobStore.java

```diff
@@ -502,6 +502,9 @@ public BlobMetadata blobMetadata(String container, String key) {
         try {
             properties = client.getProperties();
         } catch (BlobStorageException bse) {
+            if (bse.getErrorCode().equals(BlobErrorCode.BLOB_NOT_FOUND)) {
+                return null;
+            }
```

Comment:
Thanks for fixing this!  I believe that this only affected internal-only calls like the MPU ones and `handleBlobMetadata` coincidentally worked due to the top-level handling `KeyNotFoundException`.